crates/autopilot/src/domain/eth/mod.rs

```diff
@@ -0,0 +1,5 @@
+/// Domain separator used for signing.
+///
+/// https://eips.ethereum.org/EIPS/eip-712#definition-of-domainseparator
+#[derive(Debug, Clone, Copy)]
+pub struct DomainSeparator(pub [u8; 32]);
```

Comment:
nitpick: have you considered making the internal type not public and derive `Into`/`From` (`derive_more` crate)?

Reply:
Not gaining much with it, I believe. We can probably add it later if it becomes ugly.